src/setup/MetadataTask/GenerateMetadata.cs

```diff
@@ -0,0 +1,113 @@
+            Converters = { new JsonStringEnumConverter(JsonNamingPolicy.CamelCase) },
+        };
+
+        [Required]
+        public string TargetFile { get; set; }
+
+        [Required]
+        public string WixpdbFile { get; set; }
+
+        public override bool Execute()
+        {
+            var intermediate = Intermediate.Load(this.WixpdbFile);
+
+            var section = intermediate.Sections.Single();
+
+            if (section.Type != SectionType.Bundle)
+            {
+                return false;
+            }
+
+            (var metadata, var sourceLineNumber) = this.GetBundleMetadata(section);
+
+            if (metadata != null)
+            {
+                this.PopulateFileInfo(metadata);
+
+                this.SaveMetadata(metadata, sourceLineNumber);
+            }
+
+            return true;
+        }
+
+        private (Metadata, SourceLineNumber) GetBundleMetadata(IntermediateSection section)
+        {
+            var bundleSymbol = section.Symbols.OfType<WixBundleSymbol>().Single();
+
+            var metadata = new Metadata
+            {
+                Type = MetadataType.Burn,
+                Name = bundleSymbol.Name,
+                Version = bundleSymbol.Version,
+                Publisher = bundleSymbol.Manufacturer,
+                Description = "Installation for " + bundleSymbol.Name,
+                SupportUrl = bundleSymbol.HelpUrl,
+                BundleCode = bundleSymbol.BundleId,
+                UpgradeCode = bundleSymbol.UpgradeCode,
+                AboutUrl = bundleSymbol.AboutUrl,
+                Architecture = PlatformToArchitecture(bundleSymbol.Platform),
+            };
+
+            return (metadata, bundleSymbol.SourceLineNumbers);
+        }
+
+        private void PopulateFileInfo(Metadata metadata)
+        {
+            var fi = new FileInfo(this.TargetFile);
+
+            using (var sha256 = SHA256.Create())
+            using (var stream = fi.OpenRead())
+            {
+                var hash = sha256.ComputeHash(stream);
+
+                metadata.File = fi.Name;
+                metadata.Created = fi.CreationTimeUtc.ToString("O");
+                metadata.Size = fi.Length;
+                metadata.Sha256 = BitConverter.ToString(hash).Replace("-", String.Empty);
+            }
+        }
+
+        private void SaveMetadata(Metadata metadata, SourceLineNumber sourceLineNumber)
+        {
+            var metadataFilePath = Path.ChangeExtension(this.TargetFile, "metadata.json");
+
+            var json = JsonSerializer.Serialize(metadata, SerializerOptions);
+
+            File.WriteAllText(metadataFilePath, json);
+        }
+
+        private static ArchitectureType PlatformToArchitecture(Platform platform)
+        {
+            switch (platform)
+            {
+                case Platform.X86: return ArchitectureType.X86;
+                case Platform.X64: return ArchitectureType.X86;
+                case Platform.ARM64: return ArchitectureType.X86;
```

Comment:
Are they all supposed to return `ArchitectureType.X86`?

Reply:
Doh! No. Will fix.